Release 2.4.0 (#7250)
antrea-io/antrea: CHANGELOG/CHANGELOG-2.4.md

  - Clean up RBAC for FlowAggregator. ([#7125](https://github.com/antrea-io/antrea/pull/7125), [@antoninbas])
  - Use Protobuf message internally on FlowAggregator. ([#7253](https://github.com/antrea-io/antrea/pull/7253), [@antoninbas])
  - Use Protobuf / gRPC between FlowExporter and FlowAggregator by default, and allow disabling IPFIX collector via `aggregatorTransportProtocol`. ([#7264](https://github.com/antrea-io/antrea/pull/7264), [@antoninbas])
  - Push flow-aggregator image to `ghcr.io` registry. ([#7036](https://github.com/antrea-io/antrea/pull/7036), [@antoninbas])
- Log error when OVS meter drops packets, which helps to evaluate whether increasing the packetInRate configuration is needed. ([#7242](https://github.com/antrea-io/antrea/pull/7242), [@tnqn])
- Log PacketIn drops when dispatching to per-category queues to improve troubleshooting. ([#7174](https://github.com/antrea-io/antrea/pull/7174), [@tnqn])
- Increase the default packet-in rate limit to 5000. ([#7243](https://github.com/antrea-io/antrea/pull/7243), [@tnqn])

> - Sync affected groups in the Antrea Controller when a Pod goes into `Terminated` state, to ensure that the required updates are sent immediately to Agents. ([#7217](https://github.com/antrea-io/antrea/pull/7217), [@Dyanngg])

Author: The item is already included in release 2.3.1, I suppose it's better to keep the same?

> - Sync affected groups in the Antrea Controller when a Pod goes into `Terminated` state, to ensure that the required updates are sent immediately to Agents. ([#7217](https://github.com/antrea-io/antrea/pull/7217), [@Dyanngg])

Author: Updated.

- Add ICMP Rule for NodeLatencyMonitor to make it work when the Node is configured with iptables default DROP policy. ([#7011](https://github.com/antrea-io/antrea/pull/7011), [@Dhruv-J])
- Handle Pod UID updates in PodStore to handle a corner case that old and new Pods from update handler are actually different objects. ([#6964](https://github.com/antrea-io/antrea/pull/6964), [@antoninbas])
- Support configuring file permission for the Antrea CNI configuration file. ([#7098](https://github.com/antrea-io/antrea/pull/7098), [@luolanzone])
- Install iptables rules to allow WireGuard packets to ensure Antrea with WireGuard can work properly when the Node is configured with iptables default DROP policy. ([#7030](https://github.com/antrea-io/antrea/pull/7030), [@wenyingd])
- Make IPPool `prefixLength` and `gateway` immutable. ([#7186](https://github.com/antrea-io/antrea/pull/7186), [@wenqiq])
- Periodically sync permanent neighbors to ensure route correctness for Antrea host gateway interface. ([#7238](https://github.com/antrea-io/antrea/pull/7238), [@hongliangl])

> - Allow defining static MAC addresses for SecondaryInterfaces for VLAN network. ([#7137](https://github.com/antrea-io/antrea/pull/7137), [@KMAnju-2021] [@rajnkamr])

Author: Moved

- Update go-ipfix to v0.14.0. ([#7080](https://github.com/antrea-io/antrea/pull/7080), [@antoninbas])

> - Document SecondaryNetwork support for SR-IOV. ([#7076](https://github.com/antrea-io/antrea/pull/7076), [@tnqn])

Author: Updated.

### Fixed

- Enhance OVS commands for Antrea Windows to accelerate container recovery and improve robustness. ([#7228](https://github.com/antrea-io/antrea/pull/7228), [@XinShuYang])
- Configure routes via `ip route add` to avoid incorrect routes replacement when the interface is managed by a network daemon. ([#7134](https://github.com/antrea-io/antrea/pull/7134), [@luolanzone])
- Reconcile secondary network OVS ports after Agent restarts to ensure stale ports can be removed correctly. ([#6853](https://github.com/antrea-io/antrea/pull/6853), [@KMAnju-2021])

> - Rename SR-IOV VF device name back to the original one when a Pod is deleted. ([#7144](https://github.com/antrea-io/antrea/pull/7144), [@luolanzone])

Author: Hi @jianjuns I am not sure I got the question "when the secondary interface is unconfigured",  do you mean there is an issue when configuring the Pod's secondary interface continuously? then we won't restore since we rely on the stored secondary interfaces to to do the cleanup.

> - Rename SR-IOV VF device name back to the original one when a Pod is deleted. ([#7144](https://github.com/antrea-io/antrea/pull/7144), [@luolanzone])

Reviewer: I meant in addition to Pod deletion, do we also restore the VF interface name when the SR-IOV network is removed from the Pod, while the Pod is still running?

> - Rename SR-IOV VF device name back to the original one when a Pod is deleted. ([#7144](https://github.com/antrea-io/antrea/pull/7144), [@luolanzone])

Reviewer: Then should we say "after Pod or SR-IOV secondary network deletion"?

- Add `k8s.v1.cni.cncf.io/network-status` annotation to make SecondaryNetwork Pod IP visible. ([#7069](https://github.com/antrea-io/antrea/pull/7069), [@wenqiq])
- Add `protocolFilter` config to FlowExporter to filter and export flows only with the specified protocols. ([#7145](https://github.com/antrea-io/antrea/pull/7145), [@petertran-avgo])
- Add `antctl get fqdncache` sub-command to fetch the DNS mapping entries for FQDN policies. ([#6868](https://github.com/antrea-io/antrea/pull/6868), [@Dhruv-J])
- add TCP flags filter support for PacketCapture. ([#7070](https://github.com/antrea-io/antrea/pull/7070), [@AryanBakliwal])
- Add bidirectional packet capture support for PacketCapture. ([#6882](https://github.com/antrea-io/antrea/pull/6882), [@AryanBakliwal])
- Add ICMP messages filter support for PacketCapture. ([#7164](https://github.com/antrea-io/antrea/pull/7164), [@AryanBakliwal])
- Support `antctl packetcapture` sub-commands for PacketCapture. ([#6884](https://github.com/antrea-io/antrea/pull/6884), [@hangyan])
- Support enabling multicast snooping for SecondaryNetwork. ([#7200](https://github.com/antrea-io/antrea/pull/7200), [@tnqn])

> - Support `k8s.v1.cni.cncf.io/networks` annotation removal to clean up Pod's secondary interfaces for SecondaryNetwork. ([#7119](https://github.com/antrea-io/antrea/pull/7119), [@wenqiq])

Reviewer: I feel this one should be "Changed".

> - Support `k8s.v1.cni.cncf.io/networks` annotation removal to clean up Pod's secondary interfaces for SecondaryNetwork. ([#7119](https://github.com/antrea-io/antrea/pull/7119), [@wenqiq])

Reviewer 2: by removal, we also mean removing an interface definition from the annotation (in case of multiple secondary network interfaces), not just removing the annotation? it's not clear from this sentence IMO

Author: Updated, we only support removing the whole annotation or reset it to blank for now.

- Allow defining static MAC addresses for SecondaryInterfaces for VLAN network. ([#7137](https://github.com/antrea-io/antrea/pull/7137), [@KMAnju-2021] [@rajnkamr])

### Changed

> - Promote feature `AntreaIPAM` from Alpha to Beta. ([#7184](https://github.com/antrea-io/antrea/pull/7184), [@wenqiq])

Reviewer 2: maybe let's not include this if there is no clear consensus regarding ongoing discussions

Author: Removed

>   - Multiple improvements for IPFIX collector on FlowAggregator. ([#7223](https://github.com/antrea-io/antrea/pull/7223), [@antoninbas])

Author: Updated all comments @antoninbas could you take another look? thanks

>   - Add `priorityClassName` as `system-node-critical` by default for FlowAggregator. ([#7124](https://github.com/antrea-io/antrea/pull/7124), [@luolanzone])

Reviewer 2: Set `priorityClassName` to `system-node-critical`

Reviewer 2: No addressed

Author: ah, fixed.

  - Multiple improvements for IPFIX collector on FlowAggregator. ([#7223](https://github.com/antrea-io/antrea/pull/7223), [@antoninbas])
  - Add `priorityClassName` as `system-node-critical` by default for FlowAggregator. ([#7124](https://github.com/antrea-io/antrea/pull/7124), [@luolanzone])
  - Support custom ClusterIDs attached to exported flow records for FlowAggregator. ([#7197](https://github.com/antrea-io/antrea/pull/7197), [@petertran-avgo])
  - Clean up RBAC for FlowAggregator. ([#7125](https://github.com/antrea-io/antrea/pull/7125), [@antoninbas])

>   - Use Protobuf message internally on FlowAggregator. ([#7253](https://github.com/antrea-io/antrea/pull/7253), [@antoninbas])

Reviewer 2: Use Protobuf message in FlowAggregator to represent flows

  - Use Protobuf / gRPC between FlowExporter and FlowAggregator by default, and allow disabling IPFIX collector via `aggregatorTransportProtocol`. ([#7264](https://github.com/antrea-io/antrea/pull/7264), [@antoninbas])
  - Push flow-aggregator image to `ghcr.io` registry. ([#7036](https://github.com/antrea-io/antrea/pull/7036), [@antoninbas])
- Log error when OVS meter drops packets, which helps to evaluate whether increasing the packetInRate configuration is needed. ([#7242](https://github.com/antrea-io/antrea/pull/7242), [@tnqn])
- Log PacketIn drops when dispatching to per-category queues to improve troubleshooting. ([#7174](https://github.com/antrea-io/antrea/pull/7174), [@tnqn])
- Increase the default packet-in rate limit to 5000. ([#7243](https://github.com/antrea-io/antrea/pull/7243), [@tnqn])
- Sync affected groups in the Antrea Controller when a Pod goes into `Terminated` state, to ensure that the Pod is excluded from NetworkPolicy source and destination immediately. ([#7217](https://github.com/antrea-io/antrea/pull/7217), [@Dyanngg])

> - Add more configuration values to the flow-aggregator chart. ([#7138](https://github.com/antrea-io/antrea/pull/7138), [@antoninbas])

Reviewer 2: did you want to group this one with the other ones above?

- Add ICMP Rule for NodeLatencyMonitor to make it work when the Node is configured with iptables default DROP policy. ([#7011](https://github.com/antrea-io/antrea/pull/7011), [@Dhruv-J])

> - Handle Pod UID updates in PodStore to handle a corner case that old and new Pods from update handler are actually different objects. ([#6964](https://github.com/antrea-io/antrea/pull/6964), [@antoninbas])

Reviewer 2: ```suggestion
- Handle Pod UID updates in PodStore to account for the corner case where old and new Pods from update handler are actually different objects. ([#6964](https://github.com/antrea-io/antrea/pull/6964), [@antoninbas])
```

> - Support configuring file permission for the Antrea CNI configuration file. ([#7098](https://github.com/antrea-io/antrea/pull/7098), [@luolanzone])

Reviewer 2: s/permission/permissions

- Install iptables rules to allow WireGuard packets to ensure Antrea with WireGuard can work properly when the Node is configured with iptables default DROP policy. ([#7030](https://github.com/antrea-io/antrea/pull/7030), [@wenyingd])
- Make IPPool `prefixLength` and `gateway` immutable. ([#7186](https://github.com/antrea-io/antrea/pull/7186), [@wenqiq])
- Periodically sync permanent neighbors to ensure route correctness for Antrea host gateway interface. ([#7238](https://github.com/antrea-io/antrea/pull/7238), [@hongliangl])
- Rename a SR-IOV VF device, which is configured as a secondary Pod interface, back to the original name when the Pod is deleted. ([#7144](https://github.com/antrea-io/antrea/pull/7144), [@luolanzone])

> - Support removing the whole `k8s.v1.cni.cncf.io/networks` annotation or resetting it to blank to clean up Pod's secondary interfaces for SecondaryNetwork. ([#7119](https://github.com/antrea-io/antrea/pull/7119), [@wenqiq])

Reviewer 2: ```suggestion
- Support removing the whole `k8s.v1.cni.cncf.io/networks` annotation or resetting it to an empty value, which deletes the Pod's SecondaryNetwork interfaces. ([#7119](https://github.com/antrea-io/antrea/pull/7119), [@wenqiq])
```

> - Update go-ipfix to v0.14.0. ([#7080](https://github.com/antrea-io/antrea/pull/7080), [@antoninbas])

Reviewer 2: we are going to end up with go-ipfix v0.16.0 in this release, so that would be confusing

it was also meant as a bug fix for #7079, which maybe I should have made clearer

so maybe we could replace this line with "Fix invalid IPFIX UDP traffic fragmentation in the Flow Aggregator". You can keep the PR reference the same and move the entry to the Fixed` section.

### Fixed

- Enhance OVS commands for Antrea Windows to accelerate container recovery and improve robustness. ([#7228](https://github.com/antrea-io/antrea/pull/7228), [@XinShuYang])

> - Configure routes via `ip route add` to avoid incorrect routes replacement when the interface is managed by a network daemon. ([#7134](https://github.com/antrea-io/antrea/pull/7134), [@luolanzone])

Reviewer 2: route replacements
or
replacement of routes
